examples/ota_http_client.rs

```diff
@@ -0,0 +1,147 @@
+//! OTA update example.
+//!
+//! In this example, we are checking for updates from a distant server hosting the latest firmware.
+//! The hosting server is supposed to check if an update should be done or not by responding with
+//! a 200 (OK, you should update) or 304 (Not Modified, you are up to date) status code.
+//!
+//! For this example to work, you need a OTA ready partition table.
+//! You need at least 2 OTA partitions (see https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/kconfig.html#config-partition-table-type)
+//!
+//! The most common starting point for OTA is the "Factory app, two OTA definitions" layout (https://github.com/espressif/esp-idf/blob/master/components/partition_table/partitions_two_ota.csv).
+//!
+//! To use a custom partition table, download the CSV file of your choice (or create a custom one)
+//! and use either use the `--partition-table` option of `espflash`, or set this option in your `espflash.toml` file.
+//!
+//! After a successful OTA update, you will need to reset the `otadata` partition. Otherwise, the ESP
+//! will continue to boot on the second partition, while `cargo run` is flashing the first one by default.
+//! To reset the `otadata` partition, add `--erase-parts otadata` to the runner command in `.cargo/config.toml`.
+
+use anyhow::Context;
+use embedded_svc::http::client::Client as HttpClient;
+use esp_idf_svc::http::client::{EspHttpConnection, Method};
+use esp_idf_svc::io;
+use esp_idf_svc::log::EspLogger;
+use esp_idf_svc::wifi::{BlockingWifi, ClientConfiguration, Configuration, EspWifi};
+use esp_idf_svc::{eventloop::EspSystemEventLoop, nvs::EspDefaultNvsPartition};
+use esp_idf_svc::{
+    hal::peripherals::Peripherals,
+    ota::{EspOta, SlotState},
+};
+
+use log::{error, info};
+
+const VERSION: &str = "1.0.0"; // You can pull this from an environment variable at build time using env! macro.
```

Comment:
We already provide a helper macro that embeds cargo version and build information into a read only part of your application via the esp_app_desc macro defined in esp-idf-sys [here](https://github.com/esp-rs/esp-idf-sys/blob/master/src/app_desc.rs#L2). That way the version you defined in your Cargo.toml can be used and is integrated automatically

Reply:
Ooooh very cool ! I'm fixing this :-)